### ISSUE TITLE
How to configure the number of threads to use for Horovod distributed optimizer

### ISSUE THREAD

**Issue by power1628** [horovod/horovod#1556]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
How to set the number of threads to use for Horovod distributed optimizer operations.
It seems that horovod will use all available threads by default.
How can I set the the number of threads for MPI or Gloo enabled Horovod?

**Comment by tgaddair** [comment]
Hey @power1628, which threads are you referring to?  Horovod will only use a single MPI/Gloo thread for operations, though you can also run Horovod with additional MPI threads being used for other (non-Horovod) tasks. Are you referring to threads within TensorFlow?

**Comment by power1628** [comment]
hi @tgaddair  thanks for your kind reply, 'threads' here means when I'm using distributed optimizer operations, there must be some threads do the all-reduce operation and also some threads apply the gradients to parameters.
So, as you noted, the all-reduce operation is backed by MPI/Gloo threads and apply gradients operation is handled by tensorflow threads, am I right?
